harvester/sources/extraction/edurep.py

```diff
@@ -0,0 +1,319 @@
+        if publishers is None:
+            return
+        if len(publishers):
+            provider_name = publishers[0]
+        return {
+            "ror": None,
+            "external_id": None,
+            "slug": None,
+            "name": provider_name
+        }
+
+    @classmethod
+    def get_keywords(cls, node):
+        keyword_dict = node["schema:keywords"]
+        keyword_values = []
+        for keyword in keyword_dict:
+            if isinstance(keyword, str):
+                keyword_values.append("keyword is a string?")
+            elif keyword.get("@value", None) is not None:
+                keyword_values.append(keyword["@value"])
+            elif keyword.get("schema:termCode", None) is not None:
+                keyword_values.append(keyword["schema:termCode"])
+        return keyword_values
+
+    @classmethod
+    def get_organizations(cls, node):
+        root = cls.get_provider(node)
+        if root is None:
+            return
+        root["type"] = "institute"
+        return {
+            "root": root,
+            "departments": [],
+            "associates": []
+        }
+
+    @classmethod
+    def get_material_types(cls, node):
+        material_types = node.get("schema:learningResourceType", [])
+        material_values = []
+        if not material_types:
+            return []
+        for material_type in material_types:
+            if not isinstance(material_type, str):
+                material_values.append(material_type["schema:termCode"].strip())
+        return material_values
+
+    @classmethod
+    def get_publisher_date(cls, node):
+        date = node.get("schema:datePublished", None)
+        if date is None:
+            date = node.get("schema:dateModified", None)
+        if date is None:
+            date = node.get("schema:publisherDate", None)
+        if date is None:
+            return None
+        return date
+
+    @classmethod
+    def get_publisher_year(cls, node):
+
+        datetime = cls.get_publisher_date(node)
+        return date_parser(datetime).year
+
+    @classmethod
+    def get_is_restricted(cls, node):
+        return not cls.get_analysis_allowed(node)
+
+    @classmethod
+    def get_analysis_allowed(cls, node):
+        files = cls.get_files(node)
+        if not len(files):
+            return False
+        match files[0]["access_rights"], files[0]["copyright"]:
+            case "OpenAccess", _:
+                return True
+            case "RestrictedAccess", copyright:
+                return copyright and copyright not in ["yes", "unknown"] and "nd" not in copyright
+            case "ClosedAccess", _:
+                return False
+    @classmethod
+    def get_publisher(cls, node):
+        publisher = node.get("dcterms:publisher", None)
+        if publisher is None:
+            return "None"
+        else:
+            return publisher
+
+EDUREP_EXTRACTION_OBJECTIVE = {
+    # Essential NPPO properties
```

Comment:
Edurep is not NPPO (aka Publinova), but Edusources. Maybe just remove all the comments. We'll separate Edusources/Publinova specific properties another day (in code not just comments)